apps/vanilla/src/script/fetchAnime.ts

```
@@ -0,0 +1,24 @@
+
+import { Pagination } from '@js-camp/core/models/pagination';
+import { PaginationDto } from '@js-camp/core/dtos/pagination.dto';
+import { Anime } from '@js-camp/core/models/anime';
+import { AnimeDto } from '@js-camp/core/dtos/anime.dto';
+
+import { paginationMapper } from '@js-camp/core/mappers/pagination.mapper';
+import { AnimeMapper } from '@js-camp/core/mappers/anime.mapper';
+
+import { http } from '../api';
+
+/**
+ * Fetch anime data from api.
+ * @param url Api address.
+ */
+export async function fetchAnime(url: string): Promise<Pagination<Anime> | Error > {
+  try {
+    const response = await http.get<PaginationDto<AnimeDto>>(url);
+
+    return paginationMapper.fromDto(response.data, AnimeMapper.fromDto);
+  } catch (error: unknown) {
+    return new Error((error as Error).message);
+  }
```

If an error is not business-related, consider delegating error handling to function consumers instead.
Also prefer propagating errors via `throw` rather than returning them